ompi/attribute/attribute.c

```diff
@@ -381,8 +391,19 @@ static int set_value(ompi_attribute_type_t type, void *object,
 static int get_value(opal_hash_table_t *attr_hash, int key,
                      attribute_value_t **attribute, int *flag);
 static void *translate_to_c(attribute_value_t *val);
-static MPI_Fint translate_to_fortran_mpi1(attribute_value_t *val);
-static MPI_Aint translate_to_fortran_mpi2(attribute_value_t *val);
+static MPI_Aint translate_to_fortran_aint(attribute_value_t *val);
+#if SIZEOF_INT == OMPI_SIZEOF_FORTRAN_INTEGER
+static int translate_to_fortran_int(attribute_value_t *val);
+static inline MPI_Fint translate_to_fortran_mpi1(attribute_value_t *val) {
+    return translate_to_fortran_int(val);
+}
+#else
+static inline MPI_Fint translate_to_fortran_mpi1(attribute_value_t *val) {
+    return (MPI_Fint)translate_to_fortran_aint(val);
+}
+#endif
```

Comment:
This `#if` block is confusing to me for the same reason as the block above.  This stuff is so complicated that I think we should try to be as clear as possible, even if it means repeating some code.  Hence, I'd kinda prefer not to be clever and re-use the same subroutines behind the scenes, but rather have just two functions: one for INTEGER and one for Aint.  If there's some repeated code between the two, so be it.